b2share/modules/upgrade/cli.py

```diff
@@ -0,0 +1,123 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of EUDAT B2Share.
+# Copyright (C) 2017 CERN.
+#
+# B2Share is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# B2Share is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with B2Share; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""B2Share cli commands for upgrades."""
+
+
+from __future__ import absolute_import, print_function
+
+from datetime import datetime
+
+import click
+from b2share.version import __version__
+from flask import current_app
+from flask.cli import with_appcontext, ScriptInfo
+from invenio_db import db
+from invenio_indexer.cli import index as index_cmd
+from subprocess import call
+
+from click.testing import CliRunner
+
+from .models import Migration
+
+
+@click.group()
+def upgrade():
+    """B2SHARE upgrade commands."""
+
+
+@upgrade.command()
+@with_appcontext
+@click.option('-v', '--verbose', is_flag=True, default=False)
+def run(verbose):
+    """Upgrade the database, activate alembic and reindex the records."""
+    alembic = current_app.extensions['invenio-db'].alembic
+    new_migration = Migration()
+    new_migration.version = __version__
+    new_migration.date = datetime.utcnow()
+    migration_data = {'elasticsearch': False,
+                      'alembic_version': False,
+                      'alembic_upgrade': False}
+
+    if (db.engine.dialect.has_table(db.engine, 'transaction') and
+            not alembic.migration_context._has_version_table()):
+
+        # Make sure alembic version table exists in the db
+        alembic.migration_context._ensure_version_table()
+
+        # alembic_version state of B2Share v2.0.1
+        heads_v2_0_1 = ['dbdbc1b19cf2', '1ba76da94103',
```

Comment:
No need for `dbdbc1b19cf2` as it is part of other branches. See [`stamp`](http://alembic.zzzcomputing.com/en/latest/api/runtime.html#alembic.runtime.migration.MigrationContext.stamp) and [merge mechanisms](http://alembic.zzzcomputing.com/en/latest/branches.html).